src/logic/safe/utils/spendingLimits.ts

```diff
@@ -0,0 +1,307 @@
+    operation: DELEGATE_CALL,
+  }
+}
+
+type SpendingLimitTxParams = {
+  spendingLimitArgs: {
+    beneficiary: string
+    token: string
+    spendingLimitInWei: string
+    resetTimeMin: number
+    resetBaseMin: number
+  }
+  safeAddress
+}
+
+export const setSpendingLimitTx = ({
+  spendingLimitArgs: { beneficiary, token, spendingLimitInWei, resetTimeMin, resetBaseMin },
+  safeAddress,
+}: SpendingLimitTxParams): CreateTransactionArgs => {
+  const spendingLimitContract = getSpendingLimitContract()
+  const { nativeCoin } = getNetworkInfo()
+
+  return {
+    safeAddress,
+    to: SPENDING_LIMIT_MODULE_ADDRESS,
+    valueInWei: '0',
+    txData: spendingLimitContract.methods
+      .setAllowance(
+        beneficiary,
+        token === nativeCoin.address ? ZERO_ADDRESS : token,
+        spendingLimitInWei,
+        resetTimeMin,
+        resetBaseMin,
+      )
+      .encodeABI(),
+    operation: CALL,
+    notifiedTransaction: TX_NOTIFICATION_TYPES.NEW_SPENDING_LIMIT_TX,
+  }
+}
+
+export const setSpendingLimitMultiSendTx = (args: SpendingLimitTxParams): MultiSendTx => {
+  const tx = setSpendingLimitTx(args)
+
+  return {
+    to: tx.to,
+    value: Number(tx.valueInWei),
+    data: tx.txData as string,
+    operation: DELEGATE_CALL,
+  }
+}
+
+type SpendingLimitMultiSendTx = {
+  transactions: Array<MultiSendTx>
+  safeAddress: string
+}
+export const spendingLimitMultiSendTx = ({
+  transactions,
+  safeAddress,
+}: SpendingLimitMultiSendTx): CreateTransactionArgs => ({
+  safeAddress,
+  to: MULTI_SEND_ADDRESS,
+  valueInWei: '0',
+  txData: getEncodedMultiSendCallData(transactions, getWeb3()),
+  notifiedTransaction: TX_NOTIFICATION_TYPES.NEW_SPENDING_LIMIT_TX,
+  operation: DELEGATE_CALL,
+})
+
+type SpendingLimitAllowedBalance = GetTokenByAddress & {
+  tokenSpendingLimit: SpendingLimit
+}
+
+/**
+ * Calculates the remaining amount available for a particular SpendingLimit
+ * @param {string} tokenAddress
+ * @param {SpendingLimit} tokenSpendingLimit
+ * @param {List<Token>} tokens
+ * returns string
+ */
+export const spendingLimitAllowedBalance = ({
+  tokenAddress,
+  tokenSpendingLimit,
+  tokens,
+}: SpendingLimitAllowedBalance): string | number => {
+  const token = getBalanceAndDecimalsFromToken({ tokenAddress, tokens })
+
+  if (!token) {
+    return 0
+  }
+
+  const { balance, decimals } = token
+  const diff = new BigNumber(tokenSpendingLimit.amount).minus(tokenSpendingLimit.spent).toString()
+  const diffInFPNotation = fromTokenUnit(diff, decimals)
+
+  return new BigNumber(balance).gt(diffInFPNotation) ? diffInFPNotation : balance
+}
+
+type GetSpendingLimitByTokenAddress = {
+  spendingLimits?: SpendingLimit[] | null
+  tokenAddress?: string
+}
+
+/**
+ * Returns the SpendingLimit info for the specified tokenAddress
+ * @param {SpendingLimit[] | undefined | null} spendingLimits
+ * @param {string | undefined} tokenAddress
+ * @returns SpendingLimit | undefined
+ */
+export const getSpendingLimitByTokenAddress = ({
```

Comment:
Adding tests for the functions inside this file would be great, at least for the most "testable" ones